### PR TITLE
Pre-Verify for Ethereum Relay

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

File: frame/bridge/ethereum/relay/src/lib.rs [hunk 1]
```diff
@@ -1022,43 +1023,31 @@ impl<T: Send + Sync + Config> SignedExtension for CheckEthereumRelayHeaderParcel
...
-		Ok(ValidTransaction::default())
+		match call.is_sub_type() {
+			Some(Call::affirm(ref relay_header_parcel, _)) => {
+				ensure!(
+					T::RelayerGame::get_affirmed_relay_header_parcels(&RelayAffirmationId {
+						game_id: relay_header_parcel.header.number,
+						round: 0,
+						index: 0,
+					})
+					.is_none(),
```
Comment (HackFisher) [inline frame/bridge/ethereum/relay/src/lib.rs:1038]: Two relayers may affirm two different headers? It is not enough to pre validate only check the `RelayAffirmationId` as key duplication, checking `relay_header_parcel` is also required.

Comment (AurevoirXavier) [reply frame/bridge/ethereum/relay/src/lib.rs:1038]: No dispute API on bridger side. There is only one header now.

**Comment by AurevoirXavier** [comment]
@HackFisher Please review.

File: frame/bridge/ethereum/relay/src/mock.rs [hunk 2]
```diff
@@ -193,17 +193,18 @@ impl Default for ExtBuilder {
...
 	type RelayHeaderId = EthereumBlockNumber;
 	type RelayHeaderParcel = EthereumRelayHeaderParcel;
 	type RelayProofs = EthereumRelayProofs;
 
-	fn get_proposed_relay_header_parcels(
+	fn get_affirmed_relay_header_parcels(
 		_: &RelayAffirmationId<Self::RelayHeaderId>,
 	) -> Option<Vec<Self::RelayHeaderParcel>> {
-		unimplemented!()
+		Some(Default::default())
```
Comment (HackFisher) [inline frame/bridge/ethereum/relay/src/mock.rs:207]:
```suggestion
        // This is mocked for test `pre_verify_should_work`
		Some(Default::default())
```

**Review by HackFisher — APPROVED** [review]

**Pull request merged**